src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -831,10 +834,17 @@ public void Run()
             EngineController.ProcessPendingCustomNodeSyncData(scheduler);
             
             var task = new UpdateGraphAsyncTask(scheduler, verboseLogging);
+
+            // Start of graph evaluation (includes task.Initialize)
+            // Set these flags to false so that we do not get into infinite loop scenarios
+            // if listeners of EvaluationStarted decide to call RequestRun
+            RunSettings.RunEnabled = false; // Disable 'Run' button.
+            executingTask = false; // Not yet executing.
+            OnEvaluationStarted(EventArgs.Empty);
```

Comment:
moved the `OnEvaluationStarted` event call here so that it can run before `task.Initialize(EngineController, this)` (where the AST is parsed and can affect performance)

Reply:
I find it hard to reason about this change.
Is it important that `OnEvaluationStarted` used to be called *after* `silenceNodeModifications` was set true.
Do we use `OnEvaluationStarted` anywhere else internally or inside integrations?

Reply:
no need to set `silenceNodeModifications` during/before the `OnEvaluationStarted` event because we are already blocking graph runs with `ForceBlockRun`